src/app/pages/courses/courses.component.ts

```diff
@@ -1,93 +1,94 @@
 import { CoursesService } from './courses.service';
 import { LoaderBlockService, ModalService } from '../../core/components';
-import { FilterPipe } from '../../core/pipes';
 
 @Component({
   selector: 'courses',
   templateUrl: 'courses.template.html',
   styleUrls: ['courses.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  providers: [
-    FilterPipe
-  ]
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class CoursesComponent implements OnInit, OnDestroy {
+export class CoursesComponent implements OnInit {
   public coursesList: CourseItem[];
-  private originalCoursesList: CourseItem[];
-  private coursesSubscription: Subscription[];
+  public currentPage: number;
+  public itemsPerPage: number;
+  public coursesCount: number;
+  private searchPhrase: string;
 
   constructor(
       private coursesService: CoursesService,
       private cd: ChangeDetectorRef,
       private loaderBlockService: LoaderBlockService,
-      private ngZone: NgZone,
-      private modalService: ModalService,
-      private filterPipe: FilterPipe
+      private modalService: ModalService
   ) {
     this.coursesList = [];
-    this.originalCoursesList = [];
-    this.coursesSubscription = [];
+    
+    this.resetPagination();
   }
 
   ngOnInit(): void {
     console.info('CoursesComponent initialised');
 
-    this.coursesSubscription.push(
-        this.coursesService.getList().subscribe((data: CourseItem[]) => {
-          this.coursesList = data;
-          this.originalCoursesList = data;
-          this.cd.markForCheck();
-        })
-    );
+    this.getCoursesList();
+  }
+  
+  private resetPagination(): void {
+    this.currentPage = 1;
+    this.itemsPerPage = 5;
+    this.coursesCount = 0;
+  }
 
-    let time;
-    this.coursesSubscription.push(
-      this.ngZone.onUnstable.subscribe(() => time = Date.now())
-    );
+  private getCoursesList(): Subscription {
+    return this.coursesService.getList(this.currentPage, this.itemsPerPage, this.searchPhrase)
+        .subscribe((data: {courses: CourseItem[], coursesCount: number}) => {
+          if (!data) {
+            return;
+          }
 
-    this.coursesSubscription.push(
-      this.ngZone.onStable.subscribe(() => {
-        if (!time) {
-          return;
-        }
+          const { courses, coursesCount } = data;
 
-        console.log(Date.now() - time);
-      })
-    );
-  }
-
-  ngOnDestroy(): void {
-    this.coursesSubscription.forEach(subscription => subscription.unsubscribe());
+          this.coursesCount = coursesCount;
+          this.coursesList = courses;
+          this.cd.markForCheck();
+        });
   }
 
   onDeleteItem(event: any): void {
-    this.modalService.openConfirm(`Do you really want to delete #${event.id} course`).then(
+    this.modalService.openConfirm(`Do you really want to delete #${event.id} course`).subscribe(
         () => {
           this.loaderBlockService.show();
```

Comment:
Maybe it would be better to move loader into request service? So all requests will be with it